es6-lib/decoders/layer.js

```diff
@@ -206,6 +205,12 @@ class Layer extends Duplex {
       return;
     }
 
+    var geom = _.find(soqlRow, (r) => r.isGeometry);
+    if(geom && geom.vertexCount > this._maxVerticesPerRow) {
+      logger.warn(`Counted ${geom.vertexCount} vertices in ${this._count} row of layer ${this.name}`);
+      throw new Error(`Number of vertices exceeds max count of ${this._maxVerticesPerRow}`);
```

Comment:
Consider adding something like "please simplify your geometry and then resubmit".